src/iceberg/schema.h

```diff
@@ -54,13 +56,42 @@ class ICEBERG_EXPORT Schema : public StructType {
 
   [[nodiscard]] std::string ToString() const override;
 
+  [[nodiscard]] std::optional<std::reference_wrapper<const SchemaField>> GetFieldByName(
+      std::string_view name, bool case_sensitive) const override;
+
+  [[nodiscard]] std::optional<std::reference_wrapper<const SchemaField>> GetFieldByName(
+      std::string_view name) const;
+
+  [[nodiscard]] std::optional<std::reference_wrapper<const SchemaField>> GetFieldById(
+      int32_t field_id) const override;
+
   friend bool operator==(const Schema& lhs, const Schema& rhs) { return lhs.Equals(rhs); }
 
+  mutable std::unordered_map<int, size_t> id_to_index_;
+  mutable std::unordered_map<std::string, size_t> name_to_index_;
+  mutable std::unordered_map<std::string, size_t> lowercase_name_to_index_;
+  mutable std::vector<SchemaField> full_schemafield_;
```

Comment:
```suggestion
  mutable std::vector<std::reference_wrapper<const SchemaField>> full_schemafield_;
```

We may want to use const reference or pointer to avoid copy anything from the SchemaField.